libsweep/README.md

```diff
@@ -185,31 +185,75 @@ void sweep_device_start_scanning(sweep_device_s device, sweep_error_s* error)
 ```
 
 Signals the `sweep_device_s` to start scanning.
+If the motor is stationary (0Hz), will automatically set motor speed to default 5Hz.
+Will block until the the motor speed is stable (uses `sweep_device_wait_until_motor_ready` internally).
+Starts internal background thread to accumulate and queue up scans. Scans can then be retrieved using `sweep_device_get_scan`.
 In case of error a `sweep_error_s` will be written into `error`.
 
 ```c++
+void sweep_device_attempt_start_scanning(sweep_device_s device, sweep_error_s* error)
+```
+
+Non-blocking alternative to `sweep_device_start_scanning`. 
+Signals the `sweep_device_s` to start scanning. 
+Will only succeed if the motor speed is \>0Hz and stable.
+User is responsible for checking these conditions before calling. User can check that motor speed has stabilized using `sweep_device_get_motor_ready` and that motor speed is \> 0Hz using `sweep_device_get_speed`.
+Will NOT start an internal background thread. User is responsible for keeping up with incoming scans by calling `sweep_device_get_scan_direct`.
+In case of error a `sweep_error_s` will be written into `error`. This method will error on legitimate failures (ex: the motor speed is stationary or has not yet stabilized).
+
+```c++
 void sweep_device_stop_scanning(sweep_device_s device, sweep_error_s* error)
 ```
 
 Signals the `sweep_device_s` to stop scanning.
+Blocks for ~20ms to allow time for trailing data stream to collect and flush internally.
+In case of error a `sweep_error_s` will be written into `error`.
+
+```c++
+void sweep_device_wait_until_motor_ready(sweep_device_s device, sweep_error_s* error)
+```
+
+Blocks until the `sweep_device_s`'s motor speed has stabilized. This is useful whenever the motor speed is changed. The worst case is a 6 second wait for motor stabilization and calibration. If the motor is already spinning at the desired speed, this returns almost instantly. For visual reference, the blue LED on the device will blink while the motor speed is changing, and stop blinking when the speed has stabilized. If the motor speed has NOT yet stabilized, the device will respond to certain commands (`DS` or `MS`) with a status code indicating a failure to execute the command. Therefore, it is best practice to avoid this entirely by calling `sweep_device_wait_until_motor_ready` before calling a command that requires motor speed is stabilized.
+In case of error a `sweep_error_s` will be written into `error`. Will timeout.
+
+```c++
+int32_t sweep_device_get_motor_ready(sweep_device_s device, sweep_error_s* error)
+```
+
+Returns 0 if `sweep_device_s`'s motor speed has stabilized. 
+Returns 1 if not. 
```

Comment:
We require C99+ in the API, so we have `stdbool.h` - please use bool here.

https://github.com/scanse/sweep-sdk/blob/master/libsweep/include/sweep/sweep.h#L42

`bool sweep_device_is_motor_ready(..)`